squad/ci/backend/lava.py

```diff
@@ -271,16 +271,21 @@ def __get_publisher_event_socket__(self):
         return self.proxy.scheduler.get_publisher_event_socket()
 
     def __parse_results__(self, data, test_job):
+        ignore_lava_suite = self.settings.get('CI_LAVA_IGNORE_TESTSUITE', False)
+        if hasattr(test_job, 'target') and test_job.target.project_settings is not None:
+            project_settings = yaml.load(test_job.target.project_settings)
+            tmp_ignore_lava = project_settings.get('CI_LAVA_IGNORE_TESTSUITE')
+            if tmp_ignore_lava is not None:
+                ignore_lava_suite = tmp_ignore_lava
+
         infra_messages_re_list = []
-        if self.settings is not None:
-            lava_infra_error_messages = self.settings.get('CI_LAVA_INFRA_ERROR_MESSAGES', [])
-            for message_re in lava_infra_error_messages:
-                try:
-                    r = re.compile(message_re, re.I)
-                    infra_messages_re_list.append(r)
-                except re.error:
-                    # ignore incorrect expressions
-                    self.log_debug("'%s' is not a valid regex" % message_re)
+        for message_re in self.settings.get('CI_LAVA_INFRA_ERROR_MESSAGES', []):
```

Comment:
hmm, you will get AttributeError if self.settings are None or empty string.

Reply:
I changed the initial value from `self.setting` from `None` to `{}` so it wouldn't crash: https://github.com/chaws/squad/blob/add-project-setting/squad/ci/backend/null.py#L20